### PR TITLE
added empty_set and empty_array.

### PR DESCRIPTION
**Pull request by cseed**
I'm not 100% happy with this interface.  I created a thread to discuss collection constructors: http://dev.hail.is/t/collection-constructors-and-lit/72.

### CODE REVIEW

File: python/hail/expr/functions.py [hunk 1]
```diff
@@ -2661,6 +2661,30 @@ def set(collection):
...
 
+@typecheck(t=Type)
+def empty_set(t):
+    """Returns an empty set of elements of a type `t`.
+
+    Examples
+    --------
+    .. doctest::
+
+        >>> hl.eval_expr(empty_set())
```
Comment (tpoterba) [inline python/hail/expr/functions.py]: how does this work?

File: python/hail/expr/functions.py [hunk 2]
```diff
@@ -2661,6 +2661,30 @@ def set(collection):
...
+    ----------
+    t : :class:`.Type`
+        Type of the set elements.
+
+    Returns
+    -------
+    :class:`.SetExpression`
+        Empty set of elements of type `t`.
+    """
+    return filter(lambda x: False, set([null(t)]))
```
Comment (tpoterba) [inline python/hail/expr/functions.py]: could also use `range` and map, not sure which is better for now. clearly we should add this to the scala side.

File: python/hail/expr/functions.py [hunk 3]
```diff
@@ -2684,6 +2708,30 @@ def array(collection):
...
 
+@typecheck(t=Type)
+def empty_array(t):
+    """Returns an empty array of elements of a type `t`.
+
+    Examples
+    --------
+    .. doctest::
+
+        >>> hl.eval_expr(empty_array())
```
Comment (tpoterba) [inline python/hail/expr/functions.py]: again, how does this work?

File: python/hail/expr/functions.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/expr/functions.py]: ah. it doesn't. You need to add these methods to the appropriate functions.rst docs.

**Comment by tpoterba** [comment]
Maybe we should look for a different primitive that gives us this easily. `fill_array(n, value)`? `array_na(n, type)`?

File: python/hail/expr/functions.py (hunk 1 above)
Comment (cseed) [reply python/hail/expr/functions.py]: Good catch, fixed.

**Comment by cseed** [comment]
Personally I like using the existing mechanism best: add an optional type parameter on the array constructor: `hl.array([], t=hl.tint32)`.

Aside, we should also have list multiplication ...  oh, wait, that conflicts with scalar-array multiplication.  Hmm, I think, when we have ndarrays we should get rid of numeric operations on arrays.

**Pull request merged**